babyai/model.py

```diff
@@ -255,8 +255,10 @@ def forward(self, obs, memory, instr_embedding=None):
 
     def _get_instr_embedding(self, instr):
         if self.lang_model == 'gru':
-            _, hidden = self.instr_rnn(self.word_embedding(instr))
-            return hidden[-1]
+            out, _ = self.instr_rnn(self.word_embedding(instr))
+            index = -1 - (instr <= 0).sum(dim=1)
+            hidden = out[range(len(index)), index, :]
```

Comment:
if you could please use the same code in this `if` branch and in the next one for computing `lengths`, or `index`, that would be great. The same job is done in both cases, it would be great to avoid code duplication.